tests/libnet/vmibuilder.go

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package libnet
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/libnet/cloudinit"
```

Comment:
Is this the dependency you want to shed off? Wouldn't it make sense to move it into `libvmi`?

Reply:
`cloudinit` is its own builder library, not really related to libvmi.
It will be pretty odd IMO to import `libvmi/netcloudinit`, but I guess it is doable.

What was moved here is an opinionated builder, similar to the factories.

We have now unit tests importing `libvmi` and therefore also taking this one in as transitive dependency.
When and if we will manage to promote `libvmi` up, we will surely leave something behind, especially the factories (which are very specific to e2e tests). Perhaps then we should move this one back.

For now, I wanted to move odd things out.

Reply:
can you state a summary of your motivation in the commit message? It was not obvious to me.